### PR TITLE
RUBY-1925 ClientEncryption#initialize

### PR DESCRIPTION
**Pull request by egiurleo**
Use all the `Mongo::Crypt` code written so far to create an initializer for the `ClientEncryption` class, including instance variables for a key vault client and a `Mongo::Crypt::Handle` object.

### CODE REVIEW

File: lib/mongo/client_encryption.rb [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+  class ClientEncryption
+
+    # Create a new ClientEncryption object with the provided options.
+    #
+    # @param [ Mongo::Client ] client A Mongo::Client
+    #   that is connected to the MongoDB instance where the key vault
+    #   collection is stored.
+    # @param [ Hash ] options The ClientEncryption options
+    #
+    # @option options [ String ] key_vault_namespace The name of the
```
Comment (p-mongo) [inline lib/mongo/client_encryption.rb]: We use `:key_vault_namespace` etc. to indicate that options must be passed as symbols.

File: lib/mongo/client_encryption.rb [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+    #   key vault collection in the format "database.collection"
+    # @option options [ Hash ] kms_providers A hash of key management service
+    #   configuration information. Valid hash keys are :local or or :aws. There may be
```
Comment (p-mongo) [inline lib/mongo/client_encryption.rb]: `or` is duplicated.

File: spec/client_encryption_spec.rb [hunk 3]
```diff
@@ -0,0 +1,71 @@
+require 'lite_spec_helper'
+require 'mongo'
+
+require 'base64'
+
+RSpec.configure do |config|
+  config.extend(LiteConstraints)
```
Comment (p-mongo) [inline spec/client_encryption_spec.rb]: Why is this needed here?

File: lib/mongo/client_encryption.rb (hunk 1 above)
Comment (egiurleo) [reply lib/mongo/client_encryption.rb]: Thanks!

File: spec/client_encryption_spec.rb (hunk 3 above)
Comment (egiurleo) [reply spec/client_encryption_spec.rb]: It isn't! Thanks.

Comment (egiurleo) [reply spec/client_encryption_spec.rb]: It isn't! Thanks.
